packages/scene-composer/src/components/toolbars/common/styledComponents.ts

```diff
@@ -80,17 +81,19 @@ export const ToolbarItemIcon = styled.div<ToolbarItemIconProps>`
   align-items: center;
   justify-content: center;
   width: 40px;
-  height: 40px;
+  height: 100%;
 `;
 
 export const ToolbarItemMenu = styled.div<{
   isOpen?: boolean;
-  orientation?: 'horizontal' | 'vertical';
+  orientation?: ToolbarItemOrientation;
+  position: ToolbarMenuPosition;
 }>`
   display: none;
   position: absolute;
-  left: 100%;
-  top: 0;
+  left: ${({ position }) => (position === 'right' ? '100%' : position === 'bottom-left' ? '0' : undefined)};
```

Comment:
Nit: This nested ternary is a bit hard to read. Consider refactoring.

Reply:
done